components/script/dom/bindings/str.rs

```diff
@@ -306,11 +306,19 @@ impl DOMString {
     /// A valid number is the same as what rust considers to be valid,
     /// except for +1., NaN, and Infinity.
     /// https://html.spec.whatwg.org/multipage/#valid-floating-point-number
-    pub fn is_valid_number_string(&self) -> bool {
-        let input = &self.0;
-        input.parse::<f64>().ok().map_or(false, |val| {
-            !(val.is_infinite() || val.is_nan() || input.ends_with(".") || input.starts_with("+"))
-        })
+    pub fn is_valid_float_number_string(&self) -> bool {
+        parse_float_number(&self.0).is_ok()
+    }
+
+    /// https://html.spec.whatwg.org/multipage/#best-representation-of-the-number-as-a-floating-point-number
+    pub fn best_representation_of_the_float_number(&mut self) {
+        let input = &self.0.clone();
+        match parse_float_number(input) {
+            Ok(val) => {
+                self.0 = val;
+            },
+            _ => {}
+        }
```

Comment:
Maybe changing this `match` to `if let` might be better; then you don't need to use the wildcard matching here.

Reply:
Could you paste code let me know?

Reply:
I think we might be able to do this like

```rust
if let Ok(parsed_value) = parse_float_number(input) {
  self.0 = parsed_value;
}
```